File: src/main/java/com/example/demo/security/AuthController.java
package com.example.demo.security;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import com.example.demo.domain.User;
import com.example.demo.repository.UserRepository;


@RestController
@RequestMapping("/api/auth")
public class AuthController {

    private final UserRepository userRepository;
    private final JwtTokenProvider jwtTokenProvider;

    @Autowired
    public AuthController(UserRepository userRepository,
                          JwtTokenProvider jwtTokenProvider) {
        this.userRepository = userRepository;
        this.jwtTokenProvider = jwtTokenProvider;
    }

    @PostMapping("/login")
    public ResponseEntity<AuthResponse> login(@RequestBody LoginRequest req) {
        User user = userRepository.findByUsername(req.getUsername());
        if (user != null && user.getPassword().equals(req.getPassword())) {
            String token = jwtTokenProvider.generateToken(user.getUsername());
            return ResponseEntity.ok(new AuthResponse(token));
        } else {
            throw new RuntimeException("Invalid credentials");
        }
    }
}

File: src/main/kotlin/tonchovet/demo/security/AuthController.kt
package tonchovet.demo.security

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import reactor.core.publisher.Mono
import tonchovet.demo.repository.UserRepository

@RestController
@RequestMapping("/api/auth")
class AuthController(
    private val userRepository: UserRepository,
    private val jwtTokenProvider: JwtTokenProvider
) {

    @PostMapping("/login")
    fun login(@RequestBody req: LoginRequest): Mono<ResponseEntity<AuthResponse>> {
        return userRepository.findByUsername(req.username)
            .flatMap { user ->
                if (user.password == req.password) {
                    val token = jwtTokenProvider.generateToken(user.username)
                    Mono.just(ResponseEntity.ok(AuthResponse(token)))
                } else {
                    Mono.error(RuntimeException("Invalid credentials"))
                }
            }
            .switchIfEmpty(Mono.error(RuntimeException("User not found")))
    }
}

data class LoginRequest(val username: String, val password: String)
data class AuthResponse(val token: String)
